### PR TITLE
JAMES-2527 [Filter] Implement persistent storage for filter event sourcing system

### PR DESCRIPTION
**Pull request by trantienduchn**
continuing Matthieu work yesterday

### CODE REVIEW

**Comment by trantienduchn** [comment]
I would like to take a new `RuleDTO` class to replace `Rule` in `FilteringRuleSetDefineDTO` to separate DTO and POJO layer

**Comment by mbaechler** [comment]
> I would like to take a new RuleDTO class to replace Rule in FilteringRuleSetDefineDTO to separate DTO and POJO layer

great !

File: server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/filtering/FilteringRuleSetDefineDTO.java [hunk 1]
```diff
@@ -24,7 +24,6 @@
 import org.apache.james.eventsourcing.Event;
```
Comment (chibenwa) [inline server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/filtering/FilteringRuleSetDefineDTO.java]: In the class name, what does `Define` mean?

Comment (chibenwa) [reply server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/filtering/FilteringRuleSetDefineDTO.java]: Ok this answers my question, thanks :-)

**Comment by trantienduchn** [comment]
```
[4977bc6c58c92773d8201ad33dfbaf4f7a4b3ef7] [INFO] BUILD SUCCESS
[4977bc6c58c92773d8201ad33dfbaf4f7a4b3ef7] [INFO] ------------------------------------------------------------------------
[4977bc6c58c92773d8201ad33dfbaf4f7a4b3ef7] [INFO] Total time: 02:19 min (Wall Clock)
[4977bc6c58c92773d8201ad33dfbaf4f7a4b3ef7] [INFO] Finished at: 2018-08-23T08:36:06Z
[4977bc6c58c92773d8201ad33dfbaf4f7a4b3ef7] [INFO] ------------------------------------------------------------------------
```

🙄 magical build success after few minutes. test this please

**Comment by chibenwa** [comment]
> magical build success after few minutes

We have two CI working in //, strange behaviour is normal 

BTW you are looking at compilation logs, not testing logs IMO

**Comment by trantienduchn** [comment]
rebase on master

**Comment by mbaechler** [comment]
new PR here : https://github.com/linagora/james-project/pull/1686

**Pull request closed without merging**